src/averageAges.js

```diff
@@ -55,7 +77,31 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const filteredHasMother = people.filter(person => {
+    const personsMother = people.find(mother => mother.name === person.mother);
+
+    // if mother was found let's put her year of birth to persons object:
+
+    if (personsMother !== undefined) {
+      person.motherBorn = personsMother.born;
+    }
+
+    return personsMother;
+  });
+
+  const filteredIsSon = (onlyWithSon !== undefined)
+    ? filteredHasMother.filter(person => person.sex === 'm')
+    : filteredHasMother;
```

Comment:
I believe this two filters can be combined

Reply:
all done)